### PR TITLE
Rethrow ImportError, Fix example tests output check, Update bert and whisper requirements

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes
Add ImportError to list of exceptions to rethrow.  

Fix the output check for example tests. We need to check that `footprints` is not empty. Otherwise, we don't catch failed engine runs that just return empty footprints dict. 

Updated the docker and azureml requirements so that the tests are not broken anymore. 
Updated the whisper example requirement to use onnxruntime-extensions directly from pypi. Test is also updated to run `test_transcription` script also. 

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: examples/bert/docker/Dockerfile [hunk 1]
```diff
@@ -2,8 +2,9 @@ FROM mcr.microsoft.com/azureml/openmpi4.1.0-ubuntu20.04
 
 RUN pip install onnxruntime \
             datasets \
+            evaluate \
```
Comment (guotuofeng) [inline examples/bert/docker/Dockerfile:5]: I will fix in my PR https://github.com/microsoft/Olive/pull/346

**Comment by guotuofeng** [comment]
For the bert failure, I will cover in PR my https://github.com/microsoft/Olive/pull/346. For whisper test, you need wait my PR to be merged then turn on the ImportError in engine.py

**Comment by jambayk** [comment]
> For the bert failure, I will cover in PR my #346. For whisper test, you need wait my PR to be merged then turn on the ImportError in engine.py

@guotuofeng I don't think there are any conflicts between our PRs. My PR doesn't affect yours (other than the requirement files that can be rebased or merge conflict resolved. 

Checking the output for len > 0 is an extra level of securtiy on the tests. Nothing related to Olive core itself.

**Comment by guotuofeng** [comment]
> > For the bert failure, I will cover in PR my #346. For whisper test, you need wait my PR to be merged then turn on the ImportError in engine.py
> 
> @guotuofeng I don't think there are any conflicts between our PRs. My PR doesn't affect yours (other than the requirement files that can be rebased or merge conflict resolved.
> 
> Checking the output for len > 0 is an extra level of securtiy on the tests. Nothing related to Olive core itself.

The whisper test would failure with my PR since it would re-raise the ModuleNotFoundError if the onnxruntime-extension is not installed. I am supposing even if the onnxruntime-extenstion is not installed, the whisper model case would success if you don't change the utils.py. My point is we need add check at both utils.py and engine.py

**Comment by guotuofeng** [comment]
And you need turn on the ImportError in engine. That's @xiaoyu-work's comments

**Comment by jambayk** [comment]
> > > For the bert failure, I will cover in PR my #346. For whisper test, you need wait my PR to be merged then turn on the ImportError in engine.py
> > 
> > 
> > @guotuofeng I don't think there are any conflicts between our PRs. My PR doesn't affect yours (other than the requirement files that can be rebased or merge conflict resolved.
> > Checking the output for len > 0 is an extra level of securtiy on the tests. Nothing related to Olive core itself.
> 
> The whisper test would failure with my PR since it would re-raise the ModuleNotFoundError if the onnxruntime-extension is not installed. I am supposing even if the onnxruntime-extenstion is not installed, the whisper model case would success if you don't change the utils.py. My point is we need add check at both utils.py and engine.py

Yes. My point was the same. That's why I don't think my PR needs to wait for yours to merge or yours for mine. If yours merges before mine, I can add the ImportError thing you mentioned.

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**